spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/Predictor.scala

```diff
@@ -180,6 +180,41 @@ object Predictor {
     }
   }
 
+  def predictClass[T: ClassTag](dataSet: RDD[Sample[T]], batchSize: Int = -1, model: Module[T],
+    batchPerPartition: Int, featurePaddingParam: Option[PaddingParam[T]])(
+    implicit ev: TensorNumeric[T]): RDD[Sample[T]] = {
+    val shareBuffer = false
+    val modelBroad = ModelBroadcast[T]().broadcast(dataSet.sparkContext,
+      ConversionUtils.convert(model.evaluate()))
+    val partitionNum = dataSet.partitions.length
+    val totalBatch = if (batchSize > 0) {
+      require(batchSize % partitionNum == 0, s"Predictor.predict: total batch size $batchSize " +
+        s"should be divided by partitionNum ${partitionNum}")
+      batchSize
+    } else {
+      batchPerPartition * partitionNum
+    }
+    val rdd = ConversionUtils.coalesce(dataSet)
+    val otherBroad = rdd.sparkContext.broadcast(SampleToMiniBatch(
+      batchSize = totalBatch,
+      partitionNum = Some(rdd.partitions.length),
+      featurePaddingParam = featurePaddingParam))
+    val batchOut = rdd.mapPartitions { partition =>
+      val localModel = modelBroad.value()
+      val localTransformer = otherBroad.value.cloneTransformer()
+      val miniBatch = localTransformer(partition)
+      miniBatch.flatMap(batch => {
+        val output = localModel.forward(batch.getInput)
+        splitBatch(output, shareBuffer, batch.size())
+      })
+    }
+    dataSet.zip(batchOut).map(sample => {
```

Comment:
How can we make sure the order is consistent ?

Reply:
you need to resolve it

Reply:
Ok. I am fixing it. Will move the zip step into each map partitions.